catalog/plugin/catalog-plugin-querymonitor/src/test/java/ddf/catalog/querymonitor/QueryMonitorTest.java

```diff
@@ -0,0 +1,243 @@
+        assertThat(as.getCQL(), is(QUERY_TEXT));
+        assertThat(as.getSource(), is(nullValue()));
+        assertThat(as.getUniqueID(), is(uuid));
+        assertThat(as.getClientInfo(), is(CLIENT_TEXT));
+    }
+
+    @Test
+    public void testActiveSearchConstructorNullUUID() {
+        CatalogProvider mockSource = mock(CatalogProvider.class);
+        as = new ActiveSearch(QUERY_TEXT, mockSource, null, CLIENT_TEXT);
+        assertThat(as.getCQL(), is(QUERY_TEXT));
+        assertThat(as.getSource(), is(mockSource));
+        assertThat(as.getUniqueID(), instanceOf(UUID.class));
+        assertThat(as.getClientInfo(), is(CLIENT_TEXT));
+    }
+
+    @Test
+    public void testActiveSearchConstructorNullQueryString() {
+        CatalogProvider mockSource = mock(CatalogProvider.class);
+        uuid = UUID.randomUUID();
+        as = new ActiveSearch(null, mockSource, uuid, CLIENT_TEXT);
+        assertThat(as.getSource(), is(mockSource));
+        assertThat(as.getUniqueID(), is(uuid));
+        assertThat(as.getClientInfo(), is(CLIENT_TEXT));
+    }
+
+    @Test
+    public void testActiveSearchConstructorValidInputs() {
+        CatalogProvider mockSource = mock(CatalogProvider.class);
+        uuid = UUID.randomUUID();
+        as = new ActiveSearch(CQL_TEXT, mockSource, uuid, CLIENT_TEXT);
+        assertThat(as.getCQL(), is(CQL_TEXT));
+        assertThat(as.getSource(), is(mockSource));
+        assertThat(as.getUniqueID(), is(uuid));
+        assertThat(as.getClientInfo(), is(CLIENT_TEXT));
+    }
+
+    @Test
+    public void testQueryMonitorAddActiveSearchParameterNull() {
+        as = null;
+        assertThat(qmpi.addActiveSearch(as), is(false));
+    }
+
+    @Test
+    public void testQueryMonitorAddActiveSearchWithValidInput() {
+        as = new ActiveSearch(null, null, null, null);
+        assertThat(qmpi.addActiveSearch(as), is(true));
+    }
+
+    @Test
+    public void testQueryMonitorPluginImplRemoveActiveSearchUsingUUIDNullUUID()
+            throws StopProcessingException, PluginExecutionException {
+        QueryRequest mockQR = mock(QueryRequest.class);
+        propertyMap = new ConcurrentHashMap<>();
+        when(mockQR.getProperties()).thenReturn(propertyMap);
+        when(mockQR.toString()).thenReturn(QUERY_REQUEST_TEXT);
+        qmpi.setRemoveSearchAfterComplete(true);
+        qmpi.process(null, mockQR);
+        Map<UUID, ActiveSearch> activeSearchTable = qmpi.getActiveSearches();
+        if (activeSearchTable != null && !activeSearchTable.isEmpty()) {
+            Iterator it = activeSearchTable.entrySet()
+                    .iterator();
+            if (it != null && it.hasNext()) {
+                Map.Entry<UUID, ActiveSearch> tempEntry = (Map.Entry) it.next();
+                as = tempEntry.getValue();
+                UUID u = null;
+                assertThat(qmpi.removeActiveSearch(u), is(false));
+                u = as.getUniqueID();
+                qmpi.removeActiveSearch(u);
```

Comment:
It seems that the purpose of this test is to verify that `removeActiveSearch()` returns `false` when you pass `null`, so is it necessary to remove the search from the map after you've verified that behavior?